src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -3701,24 +3701,22 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <Clean Include="$(TargetFrameworkMonikerAssemblyAttributesPath)"/>
   </ItemGroup>
 
-  <Target Name="GenerateTargetFrameworkMonikerAttribute"
-          BeforeTargets="BeforeCompile"
-          DependsOnTargets="PrepareForBuild;GetReferenceAssemblyPaths"
+  <Target Name="GenerateTargetFrameworkMonikerAttribute" BeforeTargets="BeforeCompile" DependsOnTargets="PrepareForBuild;GetReferenceAssemblyPaths" AfterTargets="_WriteTargetFrameworkMonikerAttributeToFile"/>
```

Comment:
I don't think this does what you want it to do https://learn.microsoft.com/en-us/visualstudio/msbuild/target-build-order?view=vs-2022#beforetargets-and-aftertargets
> It isn't useful to specify both BeforeTargets and AfterTargets on the same target. As described in the next section, only the first target encountered will cause the new target to run.